### PR TITLE
Enable "no-floating-promises" lint-check in web package

### PR DESCRIPTION
**Pull request by ashishb**
### Description

Enable "no-floating-promises" lint-check in the web package
Fix the bugs encountered during that.

### Tested

```
packages/web $ yarn lint
```

### CODE REVIEW

File: packages/web/server/addToCRM.ts [hunk 1]
```diff
@@ -153,9 +153,9 @@ export default async function addToCRM({
     const contact = await upsertContact(preparedContact)
     const contactID = Number(contact.contact.id)
 
-    addToList({ contact: contactID, list, status: ListStatus.subscribe })
+    await addToList({ contact: contactID, list, status: ListStatus.subscribe })
```
Comment (aaronmgdr) [inline packages/web/server/addToCRM.ts]: Technically these could be a 
```
await Promise.all([
addToList({ contact: contactID, list, status: ListStatus.subscribe }),
setCustomFields(contactID, { interest, company, role, source: 'website' })
])

```

Comment (ashishb) [reply packages/web/server/addToCRM.ts]: Are both of them slow? If not, the complexity is not worth the upside.

Comment (aaronmgdr) [reply packages/web/server/addToCRM.ts]: they both depend on a third party HTTP API so slower than some things.

File: packages/web/server/addToCRM.ts [hunk 2]
```diff
@@ -125,10 +125,10 @@ const CUSTOM_FIELD_IDS = {
   source: 1,
 }
 async function setCustomFields(contactID: ActiveCampaignID, fields) {
-  Object.keys(fields).map((key) => {
+  Object.keys(fields).map(async (key) => {
```
Comment (aaronmgdr) [inline packages/web/server/addToCRM.ts]: Are you sure there aren't any weird effects of using async await in an map function?

File: packages/web/server/addToCRM.ts (hunk 1 above)
Comment (aaronmgdr) [reply packages/web/server/addToCRM.ts]: 🤷🏻‍♂️

**Comment by aaronmgdr** [comment]
Pulled down and played around with a few forms and there is a bit of a perceived decrease in snapiness since before we eagerly just said form submission was successful pretty much immediately. (although potentially untruthfully).  It might be ok to leave improving the snapiness for the future.

File: packages/web/server/addToCRM.ts (hunk 2 above)
Comment (ashishb) [reply packages/web/server/addToCRM.ts]: Afaik, it should be safe, I added `await Promise.all` to wait for all these calls to finish.

Reference: https://flaviocopes.com/javascript-async-await-array-map/

File: packages/web/server/addToCRM.ts (hunk 1 above)
Comment (ashishb) [reply packages/web/server/addToCRM.ts]: Fixed.

**Review by aaronmgdr — APPROVED** [review]
Looks good to me

**Pull request merged**